orttraining/orttraining/eager/ort_aten.cpp

```diff
@@ -682,6 +682,69 @@ at::Tensor& out) {
   return out;
 }
 
+// aten::equal(Tensor self, Tensor other) -> bool
+bool equal(
+  const at::Tensor& self,
+  const at::Tensor& other) {
+  ORT_LOG_FN(self, other);
+
+  if (
+    std::vector<at::ScalarType> supportedTypes =
+      {at::kFloat, at::kBFloat16, at::kHalf, at::kDouble, at::kLong, at::kByte, at::kInt, at::kShort, at::kBool};
+    !IsSupportedType(self, supportedTypes) ||
+    !IsSupportedType(other, supportedTypes)) {
+    return at::native::call_fallback_fn<
+      &at::native::cpu_fallback,
+      ATEN_OP(equal)>::call(self, other);
+  }
+
+  auto& invoker = GetORTInvoker(self.device());
+
+  auto ort_input_self = create_ort_value(invoker, self);
+  auto ort_input_other = create_ort_value(invoker, other);
+
+  auto* ort_tensor_self = ort_input_self.GetMutable<onnxruntime::Tensor>();
```

Comment:
Can these be switched to the non-mutable version?

Reply:
posting a new pr for just equal and i'll make this change there. Thank you @jamill